### DISCUSSION TITLE
Suggestion callback; just get singular argument

### DISCUSSION THREAD

**Discussion by Nuckerr** [Minestom/Minestom#1687] [Help] [answered]
With my suggestion callback, using both `context#getInput()` and `suggestion#getInput()` return the whole command. How do I just get my argument

**Comment by DevBlokJochem** [comment]
Hey, can you show me some code?

**Reply by Nuckerr** [reply]
```kt
setSuggestionCallback { sender, context, suggestion ->
        context.input
        suggestion.input
```
These

**Reply by Nuckerr** [reply]
On a command argument

**Comment by DevBlokJochem** [comment, marked as answer]
You can get the current argument by getting the context, or the suggestions input, and splitting the string after the last " ".
For example: 
```
gamemodes.setSuggestionCallback { sender, context, suggestion ->

            println(context.input.substringAfterLast(" "))
            println(suggestion.input.substringAfterLast(" "))
```

I have a list of all gamemode alias. I can check if the gamemode alias starts with the substring of the context/suggestion.
```
gamemodes.setSuggestionCallback { _, _, suggestion ->
            gamemodesList.forEach {
                if(it.startsWith(suggestion.input.substringAfterLast(" "))) {
                    suggestion.addEntry(SuggestionEntry(it))
                }
            }
        }
```
Now I'm checking if the gamemodes alias starts with the argument you are trying to type, and if it doesn't match, it won't show up.
